cirq-core/cirq/transformers/gauge_compiling/sqrt_cz_gauge.py

```diff
@@ -12,18 +12,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""A Gauge transformer for CZ**0.5 gate."""
+"""A Gauge transformer for CZ**±0.5 gate."""
+
+
+from typing import TYPE_CHECKING
+import numpy as np
 
 from cirq.transformers.gauge_compiling.gauge_compiling import (
     GaugeTransformer,
     GaugeSelector,
     ConstantGauge,
+    Gauge,
 )
-from cirq.ops.common_gates import CZ
-from cirq import ops
+from cirq.ops import CZ, S, X, Gateset
+
+if TYPE_CHECKING:
+    import cirq
+
+_SQRT_CZ = CZ**0.5
+_ADJ_S = S**-1
 
-SqrtCZGaugeSelector = GaugeSelector(
-    gauges=[ConstantGauge(pre_q0=ops.X, post_q0=ops.X, post_q1=ops.Z**0.5, two_qubit_gate=CZ**-0.5)]
-)
 
-SqrtCZGaugeTransformer = GaugeTransformer(target=CZ**0.5, gauge_selector=SqrtCZGaugeSelector)
+class SqrtCZGauge(Gauge):
+
+    def weight(self) -> float:
+        return 3.0
+
+    def sample(self, gate: 'cirq.Gate', prng: np.random.Generator) -> ConstantGauge:
+        if prng.choice([True, False]):
+            return ConstantGauge(two_qubit_gate=gate)
+        swap_qubits = prng.choice([True, False])
+        if swap_qubits:
+            return ConstantGauge(
+                pre_q1=X,
+                post_q1=X,
+                post_q0=S if gate == _SQRT_CZ else _ADJ_S,
+                two_qubit_gate=gate**-1,
+                swap_qubits=True,
```

Comment:
A bit perplexed - the `swap_qubits` should have no effect as the `gate**-1` will be symmetric for the q0, q1 swap.

@eliottrosenberg - is this functionality for simulation / execution of circuits that may have imperfect (ie asymmetric),  SQRT_CZ, INV_SQRT_CZ gates?